src/main/java/org/opensearch/jdbc/ConnectionImpl.java

```diff
@@ -84,7 +85,13 @@ log, new SQLNonTransientException("Could not initialize transport for the connec
             this.clusterMetadata = connectionResponse.getClusterMetadata();
             this.open = true;
         } catch (ResponseException | IOException ex) {
-            logAndThrowSQLException(log, new SQLException("Connection error "+ex.getMessage(), ex));
+            if (ex instanceof HttpException && ((HttpException) ex).getStatusCode() == 401) {
+                // 28000 is the SQLSTATE for invalid authorization specification
+                // https://docs.oracle.com/cd/E15817_01/appdev.111/b31228/appd.htm
+                logAndThrowSQLException(log, new SQLException("Connection error "+ex.getMessage(), "28000", ex));
```

Comment:
make "28000" a constant please

Reply:
Changed in https://github.com/Bit-Quill/sql-jdbc/pull/7/commits/2b562496734779a1c641d9a91f09454036076252